test/data-parse-test.py

```diff
@@ -322,5 +332,6 @@ def main() -> int:
 
     return int(is_error)
```

Comment:
nitpick: I'd just raise `SystemExit` here and forget about returning an int in `main`.

We shouldn't be catching `AssertionError`, as I mentioned above, these should be normal exceptions.

Btw, just for reference, if we were only supporting Python >=3.11, we should use `ExceptionGroup` here :smile:

Reply:
> nitpick: I'd just raise `SystemExit` here and forget about returning an int in `main`.

Sure. Do you have any reason for preferring `SystemExit` over sys.exit?

> We shouldn't be catching `AssertionError`, as I mentioned above, these should be normal exceptions.

Do you mean we should not catch `AssertionError` specifically or don't catch anything at all?
If it's the first, then sure, I'll do that in addition to changing the above AssertionError to ValueError, but if it's the second, I disagree, because the whole reason for catching is to print all of the failing files instead of exiting early.